### PR TITLE
Have PR processor automatically merge PRs

### PR DESCRIPTION
**Pull request by daviddavis**
Most of what you see in the diff is just moving around code. The only new code is in the `merge_pr` function.

### CODE REVIEW

File: .github/pr_processor.py [hunk 1]
```diff
@@ -41,78 +51,107 @@ def user_comment(issue, user=GITHUB_USER):
...
+        return
+
+    if any(commit for commit in pr.get_commits() if
+           commit.raw_data["commit"]["verification"]["verified"]):
+        merge_method = "merge"
+    else:
+        merge_method = "rebase"
+
+    pr.merge(merge_method=merge_method)
+    print(f"Merged PR {pr.number} by {merge_method} method.\n")
```
Comment (fao89) [inline .github/pr_processor.py:141]: Nice!

**Review by fao89 — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
I'm excited for this! The code looks good.

File: .github/pr_processor.py [hunk 2]
```diff
@@ -41,78 +54,109 @@ def user_comment(issue, user=GITHUB_USER):
...
+        # mergeable_state will say "blocking" if there aren't enough reviews
+        print(f"PR {pr.number} has not been reviewed, skipping merge.")
+        return
+
+    if pr.mergeable_state != "clean":
+        print(f"PR {pr.number} has state {pr.mergeable_state}, skipping merge.")
+        return
+
+    if any(commit for commit in pr.get_commits() if
+           commit.raw_data["commit"]["verification"]["verified"]):
```
Comment (ipanova) [inline .github/pr_processor.py:138]: :+1:

**Review by ipanova — APPROVED** [review]
Thank you!

**Comment by daviddavis** [comment]
Closing out due to inactivity. It seems like there are now github actions that can auto-merge as well if we want to do that.

**Pull request closed without merging**